fix: misc doc updates (#1153)
NVIDIA-NeMo/Automodel: docs/repository-structure.md

Reviewer: ```suggestion
This introductory guide presents the structure of the NeMo AutoModel repository, provides a brief overview of its parts, introduces concepts such as components and recipes, and explains how everything fits together.
```

Reviewer: ```suggestion
## What is NeMo AutoModel?
```

Reviewer: ```suggestion
NeMo AutoModel is a PyTorch library for fine-tuning and pre-training large scale models. In particular, it provides:
```

# Introduction to the NeMo AutoModel Repository

This introductory guide presents the structure of the NeMo AutoModel repository, provides a brief overview of its parts, introduces concepts such as components and recipes, and explains how everything fits together.

## What is NeMo AutoModel?

> NeMo AutoModel is a PyTorch library for fine-tuning and pre-training large scale models. In particular, it provides:

Reviewer: ```suggestion
NeMo AutoModel is a PyTorch library for fine-tuning and pretraining large-scale models. In particular, it provides:
```

Each recipe script can be executed directly using `torchrun`, for example, from the root directory:
```bash
torchrun --nproc-per-node=2 examples/llm_finetune/finetune.py -c examples/llm_finetune/llama3_2/llama3_2_1b_squad.yaml
```

> The above command will fine-tune the Llama3.2-1B model on the SQuaD dataset with two GPUs using the [`llama3_2_1b_squad.yaml`](https://github.com/NVIDIA-NeMo/Automodel/blob/824408f007c42e11471a1f9e1c975b570514d2a8/examples/llm_finetune/llama3_2/llama3_2_1b_squad.yaml) config

Reviewer: ```suggestion
The above command will fine-tune the Llama3.2-1B model on the SQuaD dataset with two GPUs using the [`llama3_2_1b_squad.yaml`](https://github.com/NVIDIA-NeMo/Automodel/blob/824408f007c42e11471a1f9e1c975b570514d2a8/examples/llm_finetune/llama3_2/llama3_2_1b_squad.yaml) config.
```

> If you want to execute on a single GPU replace `torchrun --nproc-per-node` with `python3`:

Reviewer: ```suggestion
If you want to execute on a single GPU, replace `torchrun --nproc-per-node` with `python3`:
```
